pkg/backup/item_collector.go

```diff
@@ -287,18 +292,50 @@ func (r *itemCollector) getResourceItems(log logrus.FieldLogger, gv schema.Group
 		if selector := r.backupRequest.Spec.LabelSelector; selector != nil {
 			labelSelector = metav1.FormatLabelSelector(selector)
 		}
+		listOptions := metav1.ListOptions{LabelSelector: labelSelector}
 
 		log.Info("Listing items")
-		unstructuredList, err := resourceClient.List(metav1.ListOptions{LabelSelector: labelSelector})
-		if err != nil {
-			log.WithError(errors.WithStack(err)).Error("Error listing items")
-			continue
+		unstructuredItems := make([]unstructured.Unstructured, 0)
+
+		if r.pageSize == 0 {
```

Comment:
change this to `r.pageSize > 0`?

Reply:
This branch should only trigger if pageSize is 0. This is the special "disable paging" value, mirroring the same behavior in kubectl:

```
$ kubectl get -h | grep chunk-size
      --chunk-size=500: Return large lists in chunks rather than all at once. Pass 0 to disable. This flag is beta and may change in the future.
```

The branch for a non-zero value begins at the `else` below.

Reply:
oops sorry for miscommunication but I meant:
```
if r.pageSize > 0 {
// use the pager
} else {
// zero or negative
// use the normal list-all
}
```

That was you also validate against a user sending in negative values by mistake and dont break